### ISSUE TITLE
ModuleNotFoundError: No module named 'typer.core'

### ISSUE THREAD

**Issue by tonal** [PrefectHQ/prefect#12561]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

typer v 0.12 break prefect cli

### Reproduction

```python3
$ pip install prefect
$ python -c 'import prefect.cli'
```

### Error

```python3
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/.pyenv/versions/3.12.2/envs/prog/lib/python3.12/site-packages/prefect/cli/__init__.py", line 3, in <module>
    from prefect.cli.root import app
  File "/home/user/.pyenv/versions/3.12.2/envs/prog/lib/python3.12/site-packages/prefect/cli/root.py", line 11, in <module>
    import typer.core
```

### Versions

```Text
prefect==2.16.8
typer==0.12.0
```

**Comment by desertaxle** [comment]
Thanks for the issue @tonal! I believe this PR will be fixed when the changes in [this PR](https://github.com/PrefectHQ/prefect/pull/12512/files) are released tomorrow. Until then we recommend installing a `typer` version earlier than `0.12.0`.

**Comment by desertaxle** [comment]
Resolved by 2.16.9 release
